docs/pages/machine-id/architecture.mdx

```diff
@@ -0,0 +1,46 @@
+---
+title: Machine ID Architecture
+description: How Teleport Machine ID works.
+---
+
+This section provides an overview of Teleport Machine ID's inner workings.
+
+A more in-depth specification of the workings of Machine ID can be found in
+[the Request For Discussion.](https://github.com/gravitational/teleport/blob/master/rfd/0064-bot-for-cert-renewals.md)
+
+## Role Impersonation
+
+At the core of Machine ID is the concept of role impersonation.
+
+Role Impersonation allows a user to generate credentials with a set of requested
+roles. The user does not have to hold these roles, but must have been granted 
+permission to impersonate them. The role impersonated credentials still includes
+the username of the user that generated them, so actions can be attributed to 
+the user.
+
+These credentials can then be used to complete any action that is allowed by the
+role's configured permissions. 
+
+In the case of Machine ID, the bot user is assigned a bot role, which includes 
+permissions to impersonate the roles that the user has configured.
+
+## `tbot`
+
+`tbot` is the binary that acts as the agent for Machine ID.
+
+On initial load, `tbot` uses the configured join method to obtain a set of
+credentials for the bot user from the Teleport auth service. It can then use
+these credentials to communicate with the Teleport auth service as the bot.
+
+On a configured regular period `tbot` then:
+
+- Renews these bot user credentials
+- For each configured destination:
+  - Obtains role impersonated credentials from the auth service using its
```

Comment:
What does "role impersonated" mean here?

Reply:
Hm, this is explained by the section on Role Impersonation above. Is there a way I could better make that connection obvious ?